flow/include/flow/genericactors.actor.h

```diff
@@ -1204,6 +1196,46 @@ inline Future<Void> operator||(Future<Void> const& lhs, Future<Void> const& rhs)
 	return chooseActor(lhs, rhs);
 }
 
+// wait <interval> then call what() in a loop forever
+ACTOR template <class Func>
+Future<Void> recurring(Func what, double interval, TaskPriority taskID = TaskPriority::DefaultDelay) {
+	loop choose {
+		when(wait(delay(interval, taskID))) { what(); }
+	}
+}
+
+// Invoke actorFunc() forever in a loop
+// At least wait<interval> between two actor functor invocations
+ACTOR template <class F>
+Future<Void> recurringAsync(
+    F actorFunc, // Callback actor functor
+    double interval, // Interval between two subsequent invocations of actor functor.
+    bool absoluteIntervalDelay, // Flag guarantees "interval" delay between two subequent actor functor invocations. If
+                                // not selected, guarantees provided are "at least 'interval' delay" between two
+                                // subsequent actor functor invocations, however, due to either 'poorly choose' interval
+                                // value AND/OR actor functor taking longer than expected to return, could cause actor
+                                // functor to run with no-delay
+    double initialDelay, // Initial delay interval
+    TaskPriority taskID = TaskPriority::DefaultDelay) {
+
+	wait(delay(initialDelay));
+
+	state Future<Void> val = actorFunc();
+
+	loop {
+		if (absoluteIntervalDelay) {
```

Comment:
Don't you need to move the call to actorFunc each time inside the loop for a new val? Otherwise you're just calling it once and then sleeping in the loop repeatedly.
If in your testing, the test can pass with the actorFunc only being called once, you might want to update your testing so it can catch this this bug of it not being called multiple times

Reply:
I've updated code to expose APIs to read `Counter` values, the test case is updated to `assert` regarding appropriate counter values for various scenarios:
1. Init refresh count
2. Count of refresh cycles due to `misses`
3. Count of refresh cycles by making test wait for known duration, causing `refreshCycles` to happen etc.